design/cstor/v1/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,145 @@
+
+Currently CstorPoolInstanceBlockDevice holds below configurations 
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+}
+```
+
+To get user Input proposed to add field in above struct
+
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+
+	// IsBlockDeviceExpanded informs whether blockdevice 
+	// got expanded or not
+	IsBlockDeviceExpanded bool `json:”isBlockDeviceExpanded”`
+}
+```
+
+### Steps to perform user stories
+- User has to set the value of IsBlockDeviceExpanded to true in corresponding CStorPoolInstanceBlockDevice.
+
+#### Pool Expansion Steps:
+
+Step1: Set autoexpand to on in pool by triggering `zpool set autoexpand=on <pool_name>`.
+
+Step2: Remove the partition if any exists which was created by cstor-pool container(ex: sdb9).
+
+Step3: Expand the existing partition by executing following comamnd (parted </disk> resizepart <partition_number> 100%).
+
+Step4: Inform the pool saying disk was expanded by triggering following command zpool online -e <pool_name> <disk_name>.
+
+
+### Low Level Design
+
+#### Work Flow
+- User/Admin will set the value of IsBlockDeviceExpanded to true to expand the cstor pool which was on top of blockdevice.
+- Once User/Admin updates the value CSPC controller will get update event. As part of event processing CSPC controller will identify which  CSPI needs to be updated and it will update CSPI with updated information.
+- When CSPI is updated CSPI controller will get update event and verifies which disk was expanded and performs Pool Expansion Steps only if IsBlockDeviceExpanded holds true and capacity of on blockdevice CR is greater than capacity on CStorPoolInstanceBlockDevice.
+
```

Comment:
What if capacity change happened only for a single BD in raid group of 4+1 bds  , do we have to expand in those cases ?

Reply:
Added the explanation PTAL